packages/react-radio/src/components/Radio/Radio.types.ts

```diff
@@ -1,56 +1,60 @@
 import { Label } from '@fluentui/react-label';
-import type { InputHTMLAttributes } from 'react';
 import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
 
 export type RadioSlots = {
-  root: Slot<'span'>;
+  /**
+   * The root element of the Radio.
+   *
+   * The root slot receives the `className` and `style` specified directly on the `<Radio>`.
+   * All other native props will be applied to the primary slot: `input`
+   */
+  root: NonNullable<Slot<'span'>>;
 
   /**
-   * Renders the radio indicator.
+   * The Radio's label.
    */
-  indicator: NonNullable<Slot<'div'>>;
+  label: Slot<typeof Label>;
 
   /**
-   * Hidden input that handles the checkbox's functionality.
+   * Hidden input that handles the radio's functionality.
+   *
+   * This is the PRIMARY slot: all native properties specified directly on `<Radio>` will be applied to this slot,
+   * except `className` and `style`, which remain on the root slot.
    */
   input: NonNullable<Slot<'input'>>;
 
   /**
-   * Label to be associated with RadioGroup element.
+   * A circle outline, with a filled circle icon inside when the Radio is checked.
    */
-  label: NonNullable<Slot<typeof Label>>;
+  indicator: NonNullable<Slot<'div'>>;
+};
 
+/**
+ * Radio Props
+ */
+export type RadioProps = Omit<ComponentProps<Partial<RadioSlots>, 'input'>, 'size'> & {
   /**
-   * Subtext added below label.
+   * The value of the RadioGroup when this Radio item is selected.
    */
-  subtext?: Slot<'span'>;
-};
+  value?: string;
 
-export type RadioCommons = InputHTMLAttributes<HTMLInputElement> & {
   /**
-   * Determines whether the label should be positioned bellow the indicator or next to it.
-   * @defaultvalue 'inline'
+   * The position of the label relative to the radio indicator.
+   *
+   * This defaults to `after` unless the Radio is inside a RadioGroup with `layout="horizontalStacked"`,
+   * in which case it defaults to `below`.
+   *
+   * @defaultvalue after
    */
-  labelPosition?: 'bottom' | 'inline';
+  labelPosition?: 'after' | 'below';
+
   /**
-   * Field required to pass className to container instead of input
-   * this will be solved by https://github.com/microsoft/fluentui/pull/18983
+   * Disable this Radio item.
    */
-  containerClassName?: string;
+  disabled?: boolean;
 };
 
-/**
- * Radio Props
- */
-export type RadioProps = ComponentProps<Partial<RadioSlots>> &
-  RadioCommons & {
-    /**
-     * ID of the native element that represents the checkbox.
-     */
-    id?: string;
-  };
-
 /**
  * State used in rendering Radio
  */
-export type RadioState = ComponentState<RadioSlots> & RadioCommons;
+export type RadioState = ComponentState<RadioSlots> & Required<Pick<RadioProps, 'labelPosition'>>;
```

Comment:
Is this done this way vs via a `RadioCommons` because we're planning to move away from that concept?

Reply:
More or less, yes. It seems so much cleaner this way...